tests/unit_tests/corsika/test_corsika_default_config.py

```diff
@@ -0,0 +1,84 @@
+import astropy.units as u
+import pytest
+
+from simtools.corsika.corsika_default_config import CorsikaDefaultConfig
+
+
+def compare_lists_ignoring_units(list1, list2):
+    return all(list1[i].value == pytest.approx(list2[i]) for i in range(len(list1)))
+
+
+def test_primary_setter():
+    config = CorsikaDefaultConfig()
+    config.primary = "gamma"
+    assert config.primary == "gamma"
+
+
+def test_primary_setter_invalid(caplog):
+    config = CorsikaDefaultConfig()
+    with pytest.raises(ValueError):
+        config.primary = "invalid_particle"
+        assert "Invalid primary particle: invalid_particle" in caplog.text
+
+
+def test_zenith_angle_setter():
+    config = CorsikaDefaultConfig()
+    config.zenith_angle = 30.0 * u.deg
+    assert config.zenith_angle.value == pytest.approx(30.0)
+
+
+def test_zenith_angle_setter_invalid(caplog):
+    config = CorsikaDefaultConfig()
+    with pytest.raises(ValueError):
+        config.zenith_angle = 5.0 * u.deg
+        assert "outside of the allowed interval" in caplog.text
+
+
+def test_energy_range_for_primary():
+    config = CorsikaDefaultConfig(primary="gamma", zenith_angle=40.0 * u.deg)
+    energy_range = config.energy_range_for_primary()
+    expected_energies = [6.0, 660.0]
+    compare_lists_ignoring_units(energy_range, expected_energies)
+
+    config = CorsikaDefaultConfig(primary="gamma", zenith_angle=52.0 * u.deg)
+    energy_range = config.energy_range_for_primary()
+    expected_energies = [9.24, 858.0]
+    compare_lists_ignoring_units(energy_range, expected_energies)
+
+
+def test_number_of_showers_for_primary():
+    config = CorsikaDefaultConfig(primary="gamma", zenith_angle=40.0 * u.deg)
+    number_of_showers = config.number_of_showers_for_primary()
+    assert number_of_showers == 5000
+
+
+def test_view_cone_for_primary():
+    config = CorsikaDefaultConfig(primary="gamma")
+    view_cone = config.view_cone_for_primary()
+    expected_view_cone = [0.0, 0.0]
+    compare_lists_ignoring_units(view_cone, expected_view_cone)
```

Comment:
This is trivial, but I expected to see a test of `gamma_diffuse` - as this is the closest one to `gamma`.

Reply:
No problem, I added one for gamma_diffuse as well.